docs/language/operators/README.md

```diff
@@ -301,6 +302,326 @@ conn  2018-03-24T17:15:20.607695Z CpjMvj2Cvj048u6bF1 10.164.94.120 39169     10.
+```
+
+#### Output:
+```mdtest-output
+{name:"figs",color:"brown",flavor:"plain",eater:"jessie"}
+{name:"dates",color:"brown",flavor:"sweet",note:"in season",eater:"quinn"}
+{name:"banana",color:"yellow",flavor:"sweet",eater:"quinn"}
+{name:"strawberry",color:"red",flavor:"sweet",eater:"quinn"}
+{name:"apple",color:"red",flavor:"tart",eater:"chris"}
+{name:"apple",color:"red",flavor:"tart",eater:"morgan"}
+```
+
+#### Example #5 - Streamed input
+
+In addition to the named files and Pools like we've used in the prior examples,
+Zed is also intended to work on streams of data. Here we'll combine our file
+sources into a stream that we'll pipe into `zq` via stdin. Because join requires
+two separate inputs, here we'll use the `has()` function to identify the
+records in the stream that will be treated as the left and right sides.
+
+The Zed script `inner-join-streamed.zed`:
+
+```mdtest-input inner-join-streamed.zed
+* | split (
+  => has(color) | sort flavor
+  => has(age) | sort likes
+) | inner join on flavor=likes eater:=name
+```
+
+Executing the Zed script:
+```mdtest-command
+cat fruit.ndjson people.ndjson | zq -z -I inner-join-streamed.zed -
+```
+
+#### Output:
+```mdtest-output
+{name:"figs",color:"brown",flavor:"plain",eater:"jessie"}
+{name:"banana",color:"yellow",flavor:"sweet",eater:"quinn"}
+{name:"strawberry",color:"red",flavor:"sweet",eater:"quinn"}
+{name:"dates",color:"brown",flavor:"sweet",note:"in season",eater:"quinn"}
+{name:"apple",color:"red",flavor:"tart",eater:"morgan"}
+{name:"apple",color:"red",flavor:"tart",eater:"chris"}
+```
+
+#### Example #6 - Multi-value join
+
+The equality test in a Zed join accepts only one named key from each input.
+However, joins on multiple matching values can still be performed by making the
+values available in comparable complex types, such as embedded records.
+
+To illustrate this, we'll introduce some new input data `inventory.ndjson`
+that represents a vendor's available quantity of fruit for sale. As the colors
+indicate, they separately offer both ripe and unripe fruit.
+
+```mdtest-input inventory.ndjson
+{"name":"banana","color":"yellow","quantity":1000}
+{"name":"banana","color":"green","quantity":5000}
+{"name":"strawberry","color":"red","quantity":3000}
+{"name":"strawberry","color":"white","quantity":6000}
+```
+
+Let's assume we're interested in seeing the available quantities of only the
+immediately-edible fruit/color combinations shown in our `fruit.ndjson`
+records. In the Zed script `multi-value-join.zed`, we create the keys as
+embedded records inside each input record, using the same field names and data
+types in each. We'll leave the created `fruitkey` records intact to show what
+they look like, but since it represents redundant data, in practice we'd
+typically [`drop`](#drop) it after the `join` in our Zed pipeline.
+
+```mdtest-input multi-value-join.zed
+from (
+  file fruit.ndjson => put fruitkey:={name:string(name),color:string(color)} | sort fruitkey;
```

Comment:
Is there a better way to write this out as a ZSON literal? When helping a community user recently, I'd started out with the other way I know of to do this, which here would be:

```
file fruit.ndjson => put fruitkey.name:=name,fruitkey.color:=color | sort fruitkey;
```

...but I was encouraged to try the ZSON literal approach instead. Trying to write it as a literal as I've done, it seemed to only accept it if I included the type casting. I'll bet there's a better way, but for some reason I'm not seeing it.